pkg/storage/etcd/etcd_helper.go

```diff
@@ -558,6 +576,15 @@ func (h *etcdHelper) GuaranteedUpdate(ctx context.Context, key string, ptrToType
 			ttl = *newTTL
 		}
 
+		// Since update object may have a resourceVersion set, we need to clear it here.
+		if h.versioner != nil {
+			if err := h.versioner.UpdateObject(ret, meta.Expiration, 0); err != nil {
```

Comment:
I was suggesting making the smallest change possible for 1.2 - using
versioner and not resetting self link (I don't think we have evidence
self link is causing updates to not be no-op, but if we do then yes,
we have to fix it)